python/sglang/srt/models/deepseek_v2.py

```diff
@@ -1874,7 +1892,15 @@ def forward(
             and not self.is_nextn
         )
 
-        hidden_states = self.mlp(hidden_states, forward_batch, can_fuse_mlp_allreduce)
+        # For DP with padding, reduce scatter can be used instead of all-reduce.
+        use_reduce_scatter = (
+            self.layer_communicator._communicate_summable_tensor_pair_fn
+            is CommunicateSummableTensorPairFn._scatter_hidden_states
```

Comment:
What about if `_communicate_summable_tensor_pair_fn` is `CommunicateSummableTensorPairFn._scatter` ?